### ISSUE TITLE
Revise defaults to reduce lagging on mouse drag?

### ISSUE THREAD

**Issue by albell** [openseadragon/openseadragon#1078]
The current default mouse drag settings have an appreciably long lag time before the image catches up with the pointer position. It trailing. It doesn't feel snappy. I can't think of a reason why anyone would want this behavior, ever.

I think these settings are a better default:

```
OpenSeadragon({
  id: "osd",
  prefixUrl: "//openseadragon.github.io/openseadragon/images/",
  tileSources: "//openseadragon.github.io/example-images/duomo/duomo.dzi",
  gestureSettingsMouse: {
    flickEnabled: true,
    clickToZoom: false,
    dblClickToZoom: true
  },
  zoomPerSecond: 0.5,
  zoomPerScroll: 1.3
});
```

Visible in action at: http://output.jsbin.com/hikekaroxa

This is all per the (closed) discussion at #1077.

If there's a compelling reason not to do this, you might at least add another example to the "zoom-and-pan" docs with these or similar settings.

**Comment by iangilman** [comment]
Yes, it's probably time to revisit the default settings... thanks for bringing it up!

The current settings were established many years ago (2008 maybe?) when the world was different, and they were well tuned to the time, but of course times change. That said, I think much of the OSD community (myself included) have become accustomed to the current settings. For that reason it's great to have someone new help us reevaluate, but also @albell please be aware that there is much history here, and keep an open mind. Statements like "I can't think of a reason why anyone would want this behavior, ever" come off as combative and don't help us get to the best new solution.

So... we need to figure out what the new preferred settings should be (thank you for providing a starting point), and we need to figure out how to migrate. Presumably there will still be some people who prefer the "classic" settings, so there should be an easy way to reinstate them (even if it's just a bit of documentation).

For reference, these are the current defaults:

```
OpenSeadragon({
  ...
  gestureSettingsMouse: {
    flickEnabled: false,
    clickToZoom: true,
    dblClickToZoom: false
  },
  zoomPerSecond: 1.0,
  zoomPerScroll: 1.2
});
```

I think the key bit you're interested in from the settings above is the `flickEnabled: true`. That's what gets you the immediate panning and the after-pan inertia. 

In #1077 @seanich mentioned possibly tightening the `flickMomentum` for mouse up from the default of 0.25; that should be considered as well.

I'm ambivalent on the click/double click change. I think it makes sense for Google Maps because single click means something different for them, and because people already expect zooming to be a possibility there. With OSD, we need to communicate "you can zoom this" more, and there's often nothing else to do with a click, so it's just unnecessary click inflation to require a double click. Hmm... maybe I'm not so ambivalent after all. 

`zoomPerSecond` affects how fast holding on the "Zoom In" and "Zoom Out" button zooms. As with zoom elsewhere in OSD, any number over 1 means zooming in and any number under 1 means zooming out (think of it like multiplication). The proposed change is just wrong... with a zoomPerSecond of 0.5 it means that when you hold down on the "Zoom In" button it actually zooms you out. On the other hand, the current zoomPerSecond default of 1 means that holding those buttons doesn't do anything, which is a shame. I think maybe 1.3 might be nice. 

I'm not in favor of bumping the `zoomPerScroll` up, since we've historically had trouble with scroll-zoom being way too sensitive. I think we've got that under control now, but I also think the current scroll speed is good... it's more frustrating to overshoot your mark than it is to have to scroll a little further, so it's better to err on the side of a small scroll.

Anyway, those are my thoughts... let's see what the rest of the OSD community thinks!

**Comment by albell** [comment]
@iangilman Sorry if I appeared combative. Your comments are very insightful. I appreciate that there is a deep history here. I just think UI norms have changed. Basically, we've all collectively gotten less patient with languorous UI animations.

>With OSD, we need to communicate "you can zoom this" more, and there's often nothing else to do with a click, so it's just unnecessary click inflation to require a double click. 

I think communicating "you can zoom this" should be done with UI: prominent, large tap target zoom buttons in the bottom corner of the display, where fingers can easily get to them. This is industry standard nowadays. That's a stylesheet issue. Sure, it's easy to change, but that's not a reason to get it wrong in the default.

Single-click to zoom isn't a convention I encounter very often these days. I would argue it's unexpected. FWIW I took a quick poll around the office yesterday, looking at various demos, and several people commented that they thought it was weird. Sure, it's easy to figure out once you stumble into it, but that doesn't make it good.  Again, I think touch has changed the norms around this. It's _really_ irritating when an attempted mouse drag with a tiny delta instead gets handled as a click, triggering an unwanted zoom. IMHO stationary mouse click should do nothing by default.

Thanks again. I hope this feedback is helpful.

**Comment by seanich** [comment]
> The proposed change is just wrong... with a zoomPerSecond of 0.5 it means that when you hold down on the "Zoom In" button it actually zooms you out.

Whoops! That's my bad. I put together that demo for @albell and those are the settings I'm currently using in my application. Since I have the default navigation buttons disabled it doesn't affect anything. I must have misread the documentation on that when I initially set it up.

---

I do agree that `flickEnabled` should be `true` by default as it seems to be what most people expect these days. I think the first experience most people have with a zooming and panning interface a mapping application. Most of the major mapping applications have immediate feedback panning with some momentum after releasing the pointing device (with both touch and mouse). 

The default `flickMomentum` setting (0.25) feels quite natural in my opinion, so I don't see a reason to change it.

I am actually ambivalent on the click/double click change. It makes sense to use double click in my application since single click performs other actions, but I do agree with @iangilman's point that for a more simple application single click is a sensible default.

I don't think changing the default `zoomPerScroll` value is worth doing, especially if there have been problems with increasing it in the past. I decided to change it in my usage since my team and out customers seem to prefer it a bit faster. It have much of an effect on the core experience, though.

---

Whether the defaults are changed or not, I think it would be valuable to include some "recipes" for a few different behaviours so new users can quickly get a feel for how the various options affect things.

**Comment by iangilman** [comment]
Okay, sounds like we're in agreement on flickEnabled (though I was hoping for more folks to chime in). I also think changing zoomPerSecond so it's possible to click and hold on the zoom in/out buttons would be good. Looks like a value of 1.5 is pretty good.

Maybe we should start with those changes?

@seanich I like the "recipes" idea... any thoughts about where that might live on the site or in the documentation?

Regarding the placement of the controls (moving them to the lower right), I think that's a separate issue, but I'm happy to discuss it as well. That said, I don't see a compelling argument for changing the placement. There's no reason to have it "close to your thumbs" on the desktop, and on mobile you're more likely to just pinch than to use the buttons. In fact Google Maps doesn't have zoom buttons at all on mobile. Of course the look of our buttons could definitely use some love... they're terribly out of date! So far one person has created an icon pack: 

http://openseadragon.github.io/#plugins

I'd love to see more variety there, and then we can pick a good new one to be the default.

**Comment by albell** [comment]
> Regarding the placement of the controls (moving them to the lower right), I think that's a separate issue, but I'm happy to discuss it as well. That said, I don't see a compelling argument for changing the placement. There's no reason to have it "close to your thumbs" on the desktop.

Yes, there is, for the simple reason that desktop devices can be touch devices. Many UX studies show that touch targets in top corners are less used on touch-enabled desktop. And touch targets in the center bottom are very hard to tap when used mounted on a stand, for example with a keyboard. Quoting [Luke Wroblewski ](http://www.lukew.com/ff/entry.asp?1649) from almost five years ago (and still true, in fact more than ever):

> With touch-enabled laptops, people get pretty close to the screen and use their two thumbs to tap which yields easy to hit areas in the bottom corner of the screen.
> 
> As you've hopefully observed the common pattern here is comfortable touch surfaces toward the bottom of the screen. Looking at the ergonomics of use across devices types pushes us toward the bottom, which is where we'd ideally like to place important functionality like navigation controls.

**Comment by iangilman** [comment]
Good point, and certainly a trend worth watching. I don't have enough experience with touch desktop devices to know whether tapping a button would be more appealing than pinching. I do know @msalsbery has put a lot of work into making sure OSD's touch gestures work on all touch enabled devices, and even the separate pen events (not to be confused with touch or mouse) of pen-enabled devices work. Anyway, good stuff to think about for a possible redesign of the buttons.

I see that as more of a long term change, requiring a bit more thought. Turning flickEnabled on is something we can do right away.

**Comment by ovanderzee** [comment]
Just to help out when you do not like OpenSeadragon's animation effects, reading the unit tests, you should configure:
    OpenSeadragon({
        ...
        animationTime: 0,
        springStiffness: 100,
        ...
    })
